test/components/birth-date-picker.spec.tsx

```diff
@@ -0,0 +1,141 @@
+        const onChange = sinon.spy();
+        const { select, waitForDom } = clientRenderer.render(
+            <BirthDatePicker value={new Date()} onChange={onChange} />
+        );
+
+        return waitForDom(() => {
+            expect(onChange).to.have.not.been.called;
+        });
+    });
+
+    it("Emits onChange when going from valid to valid state", function() {
+        const onChange = sinon.spy();
+        const { select, waitForDom } = clientRenderer.render(
+            <BirthDatePicker value={new Date("1986-04-26Z")} onChange={onChange} />
+        );
+
+        return waitForDom(() => {
+            change(select("BIRTH_DATE_PICKER_DAY"), "27");
+            expect(onChange).to.have.been.calledOnce.and.calledWith(new Date("1986-04-27Z"));
+        });
+    });
+
+    it("Does not emit onChange when going from valid to invalid state", function() {
+        const onChange = sinon.spy();
+        const { select, waitForDom } = clientRenderer.render(
+            <BirthDatePicker value={new Date()} onChange={onChange} />
+        );
+
+        return waitForDom(() => {
+            change(select("BIRTH_DATE_PICKER_MONTH"), "");
+            expect(onChange).to.have.not.been.called;
+        });
+    });
+
+    it("Does not emit onChange when going from invalid to invalid state", function() {
+        const onChange = sinon.spy();
+        const { select, waitForDom } = clientRenderer.render(
+            <BirthDatePicker onChange={onChange} />
+        );
+
+        return waitForDom(() => {
+            change(select("BIRTH_DATE_PICKER_MONTH"), "03");
+            expect(onChange).to.have.not.been.called;
+        });
+    });
+
+    it("Emits onChange when going from invalid to valid state", function() {
+        const onChange = sinon.spy();
+        const { select, waitForDom } = clientRenderer.render(
+            <BirthDatePicker onChange={onChange} />
+        );
+
+        return waitForDom(() => {
+            change(select("BIRTH_DATE_PICKER_YEAR"), "1986");
+            change(select("BIRTH_DATE_PICKER_MONTH"), "4");
+            change(select("BIRTH_DATE_PICKER_DAY"), "26");
+            expect(onChange).to.have.been.calledOnce.and.calledWith(new Date("1986-04-26Z"));
+        });
+    });
+
+    describe("Validates the date", function() {
+        it("Supports full range of dates", function() {
+            expect(dateFromYearMonthDay("1000", "01", "01")).to.be.instanceOf(Date);
+            expect(dateFromYearMonthDay("9999", "12", "31")).to.be.instanceOf(Date);
+        });
+
+        it("Supports leap years", function() {
+            expect(dateFromYearMonthDay("1984", "02", "29")).to.be.instanceOf(Date);
+        });
+
+        it("Allows to omit leading zeroes for month and day, but not for year", function() {
+            expect(dateFromYearMonthDay("1987", "4", "6")).to.be.instanceOf(Date);
+            expect(dateFromYearMonthDay("17", "04", "26")).to.be.instanceOf(Error);
```

Comment:
I would also check that this is the invalid date error, if it is a different error we will never know.